File: 비주얼 베이직/비베 기말/2019--1-11주차-자작함수-김유빈/2019--1-11주차-자작함수-김유빈/2019--1-11주차-자작함수-김유빈/Form3.vb
Public Class Form3
    Private Sub Form3_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        '    Call ChangeSignal(2) '노랑
    End Sub
    Private Sub ChangeSignal(ByVal Color As Byte)
        Select Case Color
            Case 1
                PictureBox1.Image = Image.FromFile(CurDir() & "\blue.ico")
            Case 2
                PictureBox1.Image = Image.FromFile(CurDir() & "\yellow.ico")
            Case 3
                PictureBox1.Image = Image.FromFile(CurDir() & "\red.ico")
            Case 4
                PictureBox1.Image = Image.FromFile(CurDir() & "\yellow.ico")

        End Select
    End Sub
    Dim Sinho_Color As Integer = 1
    Private Sub Timer1_Tick(sender As Object, e As EventArgs) Handles Timer1.Tick
        Dim Sinho_Color As Integer = 1
        PictureBox1.Image = Image.FromFile(CurDir() & "\" & Sinho_Color & ".ico")
        Sinho_Color = Sinho_Color + 1

        If Sinho_Color > 4 Then
            Sinho_Color = 1
        End If
        'Call ChangeSignal(Sinho_Color)



        'If Sinho_Color > 4 Then
        '    Sinho_Color = 1
        'End If
    End Sub
End Class

File: C#/2019-2-C샵-7주차-자작함수와신호등만들기-김유빈/2019-2-C샵-7주차-자작함수와신호등만들기-김유빈/Form3.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace _2019_2_C샵_7주차_자작함수와신호등만들기_김유빈
{
    public partial class Form3 : Form
    {
        public Form3()
        {
            InitializeComponent();
        }
        public void DisplaySinhodoong(int a)
        {
            switch (a)
            {
                case 1:
                    pictureBox1.Image = Image.FromFile
               (System.Environment.CurrentDirectory + "/red.ico");
                    break;
                case 2:
                    pictureBox1.Image = Image.FromFile
               (System.Environment.CurrentDirectory + "/yellow.ico");
                    break;
                case 3:
                    pictureBox1.Image = Image.FromFile
               (System.Environment.CurrentDirectory + "/blue.ico");
                    break;
               // case 4:
               //    pictureBox1.Image = Image.FromFile
               //(System.Environment.CurrentDirectory + "/yellow.ico");
               //     break;
            }
        }
        int Sinhodoong_Color = 1; //전역변수여야 함
        //Boolean Up_Down = false; //변수 추가할때
        private void timer1_Tick(object sender, EventArgs e)
        {
            DisplaySinhodoong(Sinhodoong_Color);
            Sinhodoong_Color++;
            if(Sinhodoong_Color > 4) //case문 4를 추가한것과 같은 맥락
            {
                DisplaySinhodoong(2);
                Sinhodoong_Color = 1;
            }//변수 하나만 사용할때

            //if(Up_Down == false) Sinhodoong_Color++;
            //if (Up_Down == true) Sinhodoong_Color--;
            //if (Sinhodoong_Color >= 3) Up_Down = true;         if (Sinhodoong_Color > 3)//3을 넘으면 다시 1로 초기화<- case를 사용할때
            //if (Sinhodoong_Color <= 1) Up_Down = false;
            //Sinhodoong_Color = 1;   case추가하여 해결할때 사용
        }
    }
    }

